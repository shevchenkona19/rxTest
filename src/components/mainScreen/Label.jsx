import React from "react";
import PropTypes from "prop-types";

export default class Label extends React.Component {
    render() {
        return (
            <div>
                {this.props.text}
            </div>
        )
    }
}

Label.propTypes = {
    text: PropTypes.string.isRequired,
};