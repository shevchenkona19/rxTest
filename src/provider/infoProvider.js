import events from "events";
import * as Rx from "rxjs";
import {combineLatest} from "rxjs";
import {auditTime} from "rxjs/operators";

const temperature = new events.EventEmitter();
const humidity = new events.EventEmitter();
const airPressure = new events.EventEmitter();

const temp = Rx.Observable.create(observable => {
    let na = "";
    try {
        temperature.on("data", data => {
            clearInterval(na);
            na = setInterval(() => observable.next("N/A"), 1000);
            return observable.next(data.toFixed(2));
        })
    } catch (e) {
        observable.error(e);
    }
});
const humid = Rx.Observable.create(observable => {
    let na = "";
    try {
        humidity.on("data", data => {
            clearInterval(na);
            na = setInterval(() => observable.next("N/A"), 1000);
            return observable.next(data.toFixed(2));
        })
    } catch (e) {
        observable.error(e);
    }
});
const pressure = Rx.Observable.create(observable => {
    let na = "";
    try {
        airPressure.on("data", data => {
            clearInterval(na);
            na = setInterval(() => observable.next("N/A"), 1000);
            return observable.next(data.toFixed(2));
        })
    } catch (e) {
        observable.error(e);
    }
});

export const mainObservable = combineLatest(temp, humid, pressure, (t, h, p) => ({temperature: t, humidity: h, pressure: p}))
    .pipe(auditTime(100));

const emitTemp = multiplier => {
    const delay = Math.random() * multiplier;
    const temp = Math.random() * 36;
    temperature.emit("data", temp);
    setTimeout(() => emitTemp(multiplier), delay);
};
const emitHumidity = multiplier => {
    const delay = Math.random() * multiplier;
    const humid = Math.random() * 100;
    humidity.emit("data", humid);
    setTimeout(() => emitHumidity(multiplier), delay);
};
const emitAirPressure = multiplier => {
    const delay = Math.random() * multiplier;
    const pressure = Math.random() * 1000;
    airPressure.emit("data", pressure);
    setTimeout(() => emitAirPressure(multiplier), delay)
};

export default function (delay) {
    if (delay === undefined) delay = 2000;
    emitAirPressure(delay);
    emitHumidity(delay);
    emitTemp(delay);
}




