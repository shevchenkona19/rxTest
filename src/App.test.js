import provide from "./provider/infoProvider";
import {mainObservable} from "./provider/infoProvider";
import React from 'react';
import renderer from 'react-test-renderer';
import Label from "./components/mainScreen/Label";

test('should show values for health monitor', () => {
    let subscription = mainObservable.subscribe(values => {
        const isRightObject = values.temperature && values.humidity && values.pressure;
        expect(isRightObject).toBe(true);
        subscription.unsubscribe();
    });
    provide();
});

test("should render right label", () => {
    const component = renderer.create(<Label text={"testing some labels"}/>);
    let tree = component.toJSON();

    expect(tree).toMatchSnapshot();
});