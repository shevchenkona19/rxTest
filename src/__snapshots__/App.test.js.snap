// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render right tree 1`] = `
<div>
  testing some labels
</div>
`;
